app/views/views.py

```diff
@@ -0,0 +1,154 @@
+import jwt
+import datetime
+from flask import Flask, request
+from flask_restplus import Api, Resource, fields
+from functools import wraps
+from ..models.signup import Register
+from ..models.signin import Login
+from ..models.ride import AddRide
+from ..db.db import getAllRides, getOneRide
+
+app = Flask(__name__)
+authorizations = {
+    'apikey': {
+        'type': 'apiKey',
+        'in': 'header',
+        'name': 'X-API-KEY'
+    }
+}
+api = Api(app, authorizations=authorizations)
+app.config.from_pyfile('../../config.py')
+
+signUp = api.model('signin', {
+    'firstName': fields.String,
+    'lastName': fields.String,
+    'email': fields.String,
+    'birthDate': fields.String,
+    'location': fields.String,
+    'password': fields.String
+    })
+signIn = api.model('signup', {
+    'email': fields.String,
+    'password': fields.String
+    })
+rides = api.model('rides', {
+    'start': fields.String,
+    'finish': fields.String,
+    'slot': fields.Integer,
+    'email': fields.String,
+    'departureDate': fields.String
+    })
+
+
+def token_required(f):
+    @wraps(f)
+    def decorated(*args, **kwargs):
+        token = None
+        if 'X-API-KEY' in request.headers:
+            token = request.headers['X-API-KEY']
+        if not token:
+            return({'result': 'token is missing'})
```

Comment:
As you are only setting the token only and only if the 'X-API-KEY' is in the headers, your tests won't pass. Currently, you are not passing the `X-API-KEY` in the header when testing. Also, you are picking the token from the X-API-KEY value